### ISSUE TITLE
[Linux] PPSSPP stuck on OpenGL

### ISSUE THREAD

**Issue by CloneMMDDCVII** [hrydgard/ppsspp#10962]
### What happens?
Despite changes in the ppsspp.ini file, when changing backend from OpenGl to Vulkan, PPSSPP does not restart, and after a manual restart, it reboots to OpenGL

### What *should* happen?
PPSSPP should initiate a Vulkan instance open startup, and upon changing backend, it should restart ppsspp.

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.

Operating System : Linux Mint 18.3
PPSSPP version :  `1.5.4-r201804231654-386c9d4-80~ubuntu16.04.1`
Hardware : Intel I7-6700HQ, Nvidia GTX1060-3GB
vulkan-utils version : `1.1.70+dfsg1-1~gpu16.04.1`

### Observed behaviour.
Dolphin Emu runs with Vulkan without issues, so my install of vulkan shouldn't be the problem.

Entering settings, PPSSPP recognises Vulkan.
`I: VulkanLoader.cpp:290: Vulkan test instance created successfully.`
`I: VulkanLoader.cpp:324: Found working Vulkan API!`

However, when starting up, it initialises a OpenGL instance.
`I: Config.cpp:440: Longest display side: -1 pixels. Choosing scale 1`
`17:23:801 Core/Config.cpp:1052 I[LOADER]: Loading controller config:`
` /home/brokenclock/.config/ppsspp/PSP/SYSTEM/controls.ini`
`
I: QtMain.cpp:385: Initializing GL graphics context`
`
I: gpu_features.cpp:142: GPU Vendor : NVIDIA Corporation ; renderer: GeForce GTX 1060/PCIe/SSE2 `
`version str: 4.5.0 NVIDIA 384.130 ; GLSL version str: 4.50 NVIDIA`
`I: QtMain.cpp:387: Using thread, starting emu thread`
`
I: NativeApp.cpp:1118: NativeResized ignored, not initialized`
`
I: NativeApp.cpp:636: NativeInitGraphics`
`
I: NativeApp.cpp:709: NativeInitGraphics completed`
`
loading control pad mappings from gamecontrollerdb.txt: SUCCESS!`
`
found control pad: X360 Controller, loading mapping: SUCCESS, mapping is:`
`030000005e0400008e02000014010000,X360
Controller,a:b0,b:b1,back:b6,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,guide:b8,leftshoulder:b4,leftstick:b9,lefttrigger:a2,leftx:a0,lefty:a1,rightshoulder:b5,rightstick:b10,righttrigger:a5,rightx:a3,righty:a4,start:b7,x:b2,y:b3,platform:Linux,`
`pad 1 has been assigned to control pad: X360 Controller`
`E: QtMain.cpp:173: Output audio samples: 1024 (requested: 2048)`
`I: GLRenderManager.cpp:166: Running first frame (0)
`

The ppsspp.ini file reads : 
`CardboardYShift = 0`
`ShowFPSCounter = 3`
`GPUBackend = 1`
`RenderingMode = 1`
`SoftwareRenderer = False`
`HardwareTransform = True`
`SoftwareSkinning = True`

### Final take on the issue
I have no real idea where the issue comes from. It's not that much on an issue to me because I can still run maxed out Emulation on a 4K screen with OpenGL, but I thought I'd report it, because Vulkan really saves kittens on lower grade hardware.

**Comment by mrcmunir** [comment]
I think not implementd switch  OpenGL <-> Vulkan on Linux QT/SDL builds. you need close PPSSPP first.

@hrydgard What do you think about this?

**Comment by hrydgard** [comment]
Yeah, it's a bug that PPSSPP is not restarted properly when you switch backends in SDL.

In the Qt build, we still only support OpenGL, need to figure out how to support Vulkan there someday..

**Comment by vnctdj** [comment]
I have no problem switching between the 2 backends on my PC with an SDL build of PPSSPP.
My system is up-to-date as of today, using nvidia-driver package.

**OS:** Debian GNU/Linux testing (buster) x86_64
**Kernel:** 4.16.0-1-amd64
**CPU:** Intel i7-6700HQ (8) @ 3.500GHz
**GPU:** NVIDIA GeForce GTX 960M
**PPSSPP:** v1.5.4-971-g97546d30f

**Comment by hrydgard** [comment]
Yeah, it works, but the UI doesn't auto-restart when you've made a backend choice.

**Comment by vnctdj** [comment]
Oh OK, then the UI is actually lying:
I change the backend, it asks me to restart, I choose "yes", the preference changes.
But indeed the restart didn't really happened, now I know why it seemed so fast!

However unlike what @CloneMMDDCVII said, the preference is saved anyway, therefore the backend change is applied on the next restart.

**Comment by hrydgard** [comment]
Yup, exactly.
